src/IECorePython/ScopedGILRelease.cpp

```diff
@@ -38,9 +38,9 @@ using namespace IECorePython;
 
 ScopedGILRelease::ScopedGILRelease()
 #if PY_VERSION_HEX >= 0x03090000
-	:	m_threadsInitialised( Py_IsInitialized() )
+	:	m_threadsInitialised( Py_IsInitialized() && PyGILState_Check() )
```

Comment:
GIL juggling is a pretty significant overhead in many places already, so this does make me wary. If you want to include this centrally here I think it would be incumbent on you to do some profiling and prove that any practical overhead is negligible.

> you used something similar here ... but haven't integrated it in the ScopedGILRelease, perhaps you remember your rationale for this decision.

I don't remember my thought processes (it may even just be that I didn't want to have to make a new cortex release), but if I were to do a retro-rationalisation it would be this...ScopedGILRelease is for releasing the GIL, and in the vast majority of use cases you _know_ you have the GIL. The only times you don't know this seem to be where you're embedded in some other system with less rigorous GIL management. Why penalise the 99% of good cases for the 1% of sloppy cases?

For the sake of argument, I guess alternatives are :

- A boolean `check = false` argument to the ScopedGILRelease constructor.
- A separate `CheckedGILRelease` class.

If we do go with the approach in this PR (after verifying the overhead), I'd note that `m_threadsInitialised` is no longer an accurate name for the variable.

Reply:
I implemented the suggestion of a  `CheckedGILRelease` class here 2b5e144bd8fef892bae75de36bcf1f7ca19447d6